server/src/com/thoughtworks/go/server/service/ArtifactsService.java

```diff
@@ -390,7 +401,7 @@ private boolean deleteFile(File file) {
     }
 
     public void appendToConsoleLog(JobIdentifier jobIdentifier, String text) throws IllegalArtifactLocationException, IOException {
-        File file = findArtifact(jobIdentifier, ArtifactLogUtil.getConsoleLogOutputFolderAndFileName());
```

Comment:
Yeah, changing the order will unintentionally break performance. I think we should either have a test explaining and verifying the order or should have methods that explain it.